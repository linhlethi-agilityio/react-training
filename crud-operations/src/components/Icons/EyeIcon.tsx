import { SVGProps } from 'react';

export const EyeIcon = ({ width = 15, height = 12, fill = '#FEAF00', ...rest }: SVGProps<SVGSVGElement>) => (
  <svg width={width} height={height} viewBox={`0 0 ${width} ${height}`} {...rest}>
    <path
      fill={fill}
      d="M7.5 6.875c.442 0 .866-.184 1.178-.513.313-.328.489-.773.489-1.237 0-.464-.176-.91-.489-1.237A1.627 1.627 0 0 0 7.5 3.375c-.026 0-.049.007-.074.008.08.232.096.484.045.725a1.319 1.319 0 0 1-.333.637 1.221 1.221 0 0 1-.607.35c-.23.053-.469.036-.69-.048 0 .027-.008.051-.008.078 0 .23.043.457.127.67.084.212.207.405.361.567.313.329.737.513 1.179.513Zm7.41-1.274C13.496 2.708 10.7.75 7.5.75 4.299.75 1.502 2.709.09 5.601a.92.92 0 0 0 0 .798C1.504 9.292 4.3 11.25 7.5 11.25c3.201 0 5.998-1.959 7.41-4.851a.922.922 0 0 0 0-.798ZM7.5 1.625c.66 0 1.304.205 1.852.59.548.384.975.931 1.228 1.57.252.64.318 1.344.19 2.023A3.557 3.557 0 0 1 9.856 7.6c-.466.49-1.06.823-1.707.958a3.186 3.186 0 0 1-1.926-.2A3.374 3.374 0 0 1 4.728 7.07a3.626 3.626 0 0 1-.561-1.944 3.595 3.595 0 0 1 .977-2.474A3.259 3.259 0 0 1 7.5 1.625Zm0 8.75C4.704 10.375 2.15 8.699.833 6a7.877 7.877 0 0 1 3.38-3.536 4.474 4.474 0 0 0-.88 2.661c0 1.16.44 2.273 1.22 3.094A4.069 4.069 0 0 0 7.5 9.5c1.105 0 2.165-.46 2.946-1.281a4.488 4.488 0 0 0 1.22-3.094 4.474 4.474 0 0 0-.879-2.661A7.878 7.878 0 0 1 14.167 6c-1.316 2.699-3.871 4.375-6.667 4.375Z"
    />
  </svg>
);
