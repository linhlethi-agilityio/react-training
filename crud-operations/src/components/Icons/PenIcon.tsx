import { SVGProps } from 'react';

export const PenIcon = ({ width = 19, height = 19, fill = '#FEAF00', ...rest }: SVGProps<SVGSVGElement>) => (
  <svg width={width} height={height} viewBox={`0 0 ${width} ${height}`} {...rest}>
    <path
      fill={fill}
      d="M18.303 2.088 16.911.696A2.366 2.366 0 0 0 15.232 0c-.608 0-1.216.232-1.68.695L.476 13.772l-.471 4.239a.89.89 0 0 0 .983.983l4.235-.468 13.08-13.08a2.374 2.374 0 0 0 0-3.358ZM4.678 17.392l-3.452.383.384-3.457 9.793-9.793 3.072 3.071-9.797 9.796ZM17.464 4.607l-2.15 2.15-3.071-3.072 2.15-2.15a1.18 1.18 0 0 1 .84-.347c.316 0 .614.123.839.347l1.392 1.392a1.19 1.19 0 0 1 0 1.68Z"
    />
  </svg>
);
