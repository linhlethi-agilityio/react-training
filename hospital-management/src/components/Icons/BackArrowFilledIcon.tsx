import { memo } from 'react';
import { IIconProps } from './types';

const BackArrowFilledIcon = ({ size, color, ...props }: IIconProps) => {
  const defaultColor = '#000100';
  const originHeight = 40;
  const originWidth = 24;
  const heightRatio = originHeight / originWidth;

  return (
    <svg
      width={size}
      height={size * heightRatio}
      viewBox={`0 0 ${originWidth} ${originHeight}`}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M11.818 18.637c-.28.447-.47.944-.556 1.465.007.506.206.991.556 1.357l11.034 11.464a4.232 4.232 0 0 1 .02 5.774l-.056.062a3.916 3.916 0 0 1-5.708.014L.574 21.547A2.157 2.157 0 0 1 0 20.102c.088-.55.283-1.076.574-1.55l16.534-17.33a3.918 3.918 0 0 1 5.708.024l.056.072a4.266 4.266 0 0 1-.02 5.8l-11.034 11.52Z"
        fill={color ?? defaultColor}
      />
    </svg>
  );
};

BackArrowFilledIcon.defaultProps = {
  size: 24,
  color: '#000100',
};

export default memo(BackArrowFilledIcon);
