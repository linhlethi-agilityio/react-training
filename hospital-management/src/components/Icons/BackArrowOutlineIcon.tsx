import { memo } from 'react';
import { IIconProps } from './types';

const BackArrowOutlineIcon = ({ size, color, ...props }: IIconProps) => {
  const defaultColor = '#000';
  const originHeight = 40;
  const originWidth = 40;
  const heightRatio = originHeight / originWidth;

  return (
    <svg
      width={size}
      height={size * heightRatio}
      viewBox={`0 0 ${originWidth} ${originHeight}`}
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="m14 19.053 8.474-8.474a1.018 1.018 0 0 1 1.473 0 1.018 1.018 0 0 1 0 1.474l-7.736 7.736 8.157 8.158a1.017 1.017 0 0 1 0 1.474 1.018 1.018 0 0 1-1.473 0L14 20.526a1.017 1.017 0 0 1 0-1.473ZM0 20C0 8.947 8.947 0 20 0s20 8.947 20 20-8.947 20-20 20S0 31.053 0 20Zm2.105 0c0 9.895 8 17.895 17.895 17.895s17.895-8 17.895-17.895S29.895 2.105 20 2.105 2.105 10.105 2.105 20Z"
        fill={color ?? defaultColor}
      />
    </svg>
  );
};

BackArrowOutlineIcon.defaultProps = {
  size: 40,
  color: '#000',
};

export default memo(BackArrowOutlineIcon);
