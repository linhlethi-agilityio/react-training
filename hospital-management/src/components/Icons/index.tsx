export { default as DownloadIcon } from './DownloadIcon';

export { default as PenIcon } from './PenIcon';

export { default as FacebookV1Icon } from './FacebookV1Icon';

export { default as FacebookV2Icon } from './FacebookV2Icon';

export { default as InstagramIcon } from './InstagramIcon';

export { default as LinkedinIcon } from './LinkedinIcon';

export { default as LocationIcon } from './LocationIcon';

export { default as ForwardArrowFilledIcon } from './ForwardArrowFilledIcon';

export { default as ForwardArrowOutlineIcon } from './ForwardArrowOutlineIcon';

export { default as BackArrowFilledIcon } from './BackArrowFilledIcon';

export { default as BackArrowOutlineIcon } from './BackArrowOutlineIcon';

export { default as SearchIcon } from './SearchIcon';

export { default as TwitterIcon } from './TwitterIcon';

export { default as UserIcon } from './UserIcon';

export { default as RightArrowAltIcon } from './RightArrowAltIcon';
