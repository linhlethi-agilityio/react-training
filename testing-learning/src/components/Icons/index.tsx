export { default as BackArrowIcon } from './BackArrowIcon';

export { default as ShoppingCartIcon } from './ShoppingCartIcon';

export { default as CloseIcon } from './CloseIcon';

export { default as SearchIcon } from './SearchIcon';
