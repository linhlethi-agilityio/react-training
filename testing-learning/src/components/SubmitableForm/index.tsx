import StyledInput from '@components/Input';
import { FormEvent } from 'react';
import { transformToStyleValue } from 'src/helper/transformNumber';
import styled from 'styled-components';
import { IFormControlProps } from './types';

const SubmitableForm = (props: IFormControlProps) => {
  const {
    className,
    IconComponent,
    onChange,
    onSubmit,
    placeholderInput,
    iconSize,
    inputVariant,
    value,
  } = props;

  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    onSubmit?.();
  };

  return (
    <form className={className} onSubmit={handleSubmit}>
      <StyledInput
        onChange={onChange}
        inputWidth="full"
        variant={inputVariant}
        placeholder={placeholderInput}
        value={value}
      />
      <button type="submit">{IconComponent && <IconComponent size={iconSize!} />}</button>
    </form>
  );
};

const StyledSubmitableForm = styled(SubmitableForm)`
  position: relative;
  width: ${(props) => transformToStyleValue(props.formWidth)};

  input {
    padding-bottom: 12px;
  }

  button {
    border: none;
    background: none;
    cursor: pointer;
    position: absolute;
    right: 0;
    top: 6px;
  }
`;

export { StyledSubmitableForm };
