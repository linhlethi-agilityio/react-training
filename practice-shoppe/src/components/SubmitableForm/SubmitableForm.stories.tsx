import { ComponentMeta, ComponentStory } from '@storybook/react';
import { StyledSubmitableForm } from '@components/SubmitableForm';
import { IFormControlProps } from './types';
import { BackArrowIcon } from '@components/Icons';
import { action } from '@storybook/addon-actions';

export default {
  title: 'SubmitableForm',
  component: StyledSubmitableForm,
} as ComponentMeta<typeof StyledSubmitableForm>;

const Template: ComponentStory<typeof StyledSubmitableForm> = (args) => (
  <StyledSubmitableForm {...args} />
);

export const DefaultSubmitableForm = Template.bind({});
const defaultSubmitableForm: IFormControlProps = {
  formWidth: 396,
  placeholderInput: 'Give an email, get the newsletter.',
  IconComponent: BackArrowIcon,
  iconSize: 25,
  inputVariant: 'secondary',
  onSubmit: () => action('onSubmit')(),
  onChange: () => action('onChange')(),
  value: '',
};
DefaultSubmitableForm.args = defaultSubmitableForm;
