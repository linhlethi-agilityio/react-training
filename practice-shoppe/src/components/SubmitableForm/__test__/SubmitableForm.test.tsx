import { RenderResult, fireEvent, render } from '@testing-library/react';
import { StyledSubmitableForm } from '@components/SubmitableForm';
import { BackArrowIcon } from '@components/Icons';

describe('testing SubmitableForm component', () => {
  let wrapper: RenderResult;

  const onChange = jest.fn();
  const onSubmit = jest.fn();

  beforeEach(() => {
    wrapper = render(
      <StyledSubmitableForm
        placeholderInput="Email"
        inputVariant="secondary"
        value="value"
        onChange={onChange}
        onSubmit={onSubmit}
        IconComponent={BackArrowIcon}
      />,
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = wrapper;

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render SubmitableForm correctly', () => {
    const { getByRole } = wrapper;

    const input = getByRole('textbox') as HTMLInputElement;
    const button = getByRole('button') as HTMLButtonElement;

    expect(input.value).toBe('value');
    expect(button).toBeTruthy();
  });

  it('should call onChange when change input', () => {
    const { getByRole } = wrapper;

    const input = getByRole('textbox') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'new value' } });
    expect(onChange).toHaveBeenCalled();
  });

  it('should call onSubmit when click button', () => {
    const { getByRole } = wrapper;

    const button = getByRole('button') as HTMLButtonElement;

    fireEvent.click(button);
    expect(onSubmit).toBeCalledTimes(1);
  });
});
