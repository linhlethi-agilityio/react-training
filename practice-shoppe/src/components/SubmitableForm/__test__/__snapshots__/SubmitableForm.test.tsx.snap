// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SubmitableForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="sc-gueYoa beEJqU"
  >
    <input
      class="sc-beqWaB cNLbQU"
      placeholder="Email"
      value="value"
    />
    <button
      type="submit"
    >
      <svg
        fill="none"
        height="9"
        viewBox="0 0 25 9"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 4.991v-1.25c0-.345.28-.625.625-.625h19.381v-2.5A.625.625 0 0 1 21.07.18l3.751 3.75c.24.247.24.64 0 .887l-3.751 3.75a.625.625 0 0 1-1.063-.45v-2.5H.626A.625.625 0 0 1 0 4.991Z"
          fill="#707070"
        />
      </svg>
    </button>
  </form>
</DocumentFragment>
`;
